pkg/sensors/sensors_windows.go

```diff
@@ -0,0 +1,314 @@
+	// sensors that support policyfilter can use the filterID to implement filtering.
+	// sensors that do not support policyfilter need to return an error if filterID != policyfilter.NoFilterID
+	PolicyHandler(policy tracingpolicy.TracingPolicy, filterID policyfilter.PolicyID) (SensorIface, error)
+}
+
+type probeLoader interface {
+	LoadProbe(args LoadProbeArgs) error
+}
+
+var (
+	// list of registered policy handlers, see RegisterPolicyHandlerAtInit()
+	registeredPolicyHandlers = map[string]policyHandler{}
+	// list of registers loaders, see registerProbeType()
+	registeredProbeLoad = map[string]probeLoader{}
+	standardTypes       = map[string]func(string, *program.Program, int) error{
+		"windows": program.LoadWindowsProgram,
+	}
+)
+
+// RegisterPolicyHandlerAtInit registers a handler for a tracing policy.
+func RegisterPolicyHandlerAtInit(name string, h policyHandler) {
+	if _, exists := registeredPolicyHandlers[name]; exists {
+		panic(fmt.Sprintf("RegisterPolicyHandlerAtInit called, but %s is already registered", name))
+	}
+	registeredPolicyHandlers[name] = h
+}
+
+// RegisterProbeType registers a handler for a probe type string
+//
+// This function is meant to be called in an init() by sensors that
+// need extra logic when loading a specific probe type.
+func RegisterProbeType(probeType string, s probeLoader) {
+	logger.GetLogger().WithField("probeType", probeType).WithField("sensors", s).Debug("Registered probe type")
+	if _, exists := registeredProbeLoad[probeType]; exists {
+		panic(fmt.Sprintf("RegisterProbeType called, but %s is already registered", probeType))
+	}
+	registeredProbeLoad[probeType] = s
+}
+
+// LoadProbeArgs are the args to the LoadProbe function.
+type LoadProbeArgs struct {
+	BPFDir           string
+	Load             *program.Program
+	Version, Verbose int
+}
+
+func GetMergedSensorFromParserPolicy(tp tracingpolicy.TracingPolicy) (SensorIface, error) {
+	// NB: use a filter id of 0, so no filtering will happen
+	sis, err := SensorsFromPolicy(tp, policyfilter.NoFilterID)
+	if err != nil {
+		return nil, err
+	}
+	sensors := make([]*Sensor, 0, len(sis))
+	for _, si := range sis {
+		s, ok := si.(*Sensor)
+		if !ok {
+			return nil, fmt.Errorf("cannot merge sensor of type %T", si)
+		}
+		sensors = append(sensors, s)
+	}
+
+	return SensorCombine(tp, tp.TpName(), sensors...), nil
+}
+
+func progsAdd(progs []*program.Program) {
+	allProgramsMutex.Lock()
+	defer allProgramsMutex.Unlock()
+
+	allPrograms = append(allPrograms, progs...)
+}
+
+func progsCleanup() {
+	allProgramsMutex.Lock()
+	defer allProgramsMutex.Unlock()
+
+	progs := []*program.Program{}
+
+	for _, p := range allPrograms {
+		if p.LoadState.IsLoaded() {
+			progs = append(progs, p)
+		}
+	}
+
+	allPrograms = progs
+}
+
+func AllPrograms() []*program.Program {
+	return append([]*program.Program{}, allPrograms...)
+}
+
+// sortSensors sort the sensors to enforce orderging constrains
+func sortSensors(sensors []SensorIface) {
```

Comment:
Is this windows specific? Should we maybe move it to a common file for both archs?